packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -623,7 +627,60 @@ export interface SdkPagingServiceMethod<TServiceOperation extends SdkServiceOper
 }
 
 interface SdkLroServiceMethodOptions {
+  /**
+   * @deprecated This property will be removed in future releases. Use `lroMetadata` for synthesized LRO metadata. If you still want to access primitive LRO info, use `lroMetadata.__raw`.
+   */
   __raw_lro_metadata: LroMetadata;
+  lroMetadata: SdkLroServiceMetadata;
+}
+
+/**
+ * Long running operation metadata.
+ */
+export interface SdkLroServiceMetadata {
+  /** LRO metadata from TypeSpec core library */
+  __raw: LroMetadata;
+
+  /** Legacy `finalStateVia` value */
+  finalStateVia: FinalStateValue;
+  /** Polling step metadata */
+  pollingStep: SdkLroServicePollingStep;
+  /** Final step metadata */
+  finalStep?: SdkLroServiceFinalStep;
+  /** Synthesized final response metadata */
+  finalResponse?: SdkLroServiceFinalResponse;
+}
+
+/**
+ * Long running operation polling step metadata.
+ */
+export interface SdkLroServicePollingStep {
+  /** Response body type */
+  responseBody?: SdkModelType;
+}
+
+/**
+ * Long running operation final step metadata.
+ */
+export interface SdkLroServiceFinalStep {
+  /** Final step kind */
+  kind:
```

Comment:
Would it be better to make it as enum instead of interface with `kind`? Because in TCGC's type convention, seems `kind` is used for describing a type, like `SdkType`, but I don't see any interface extends this `SdkLroServiceFinalStep`.

Also could we add some document to say what does each kind of final step mean, and what does corresponding `finalResponse` look like?

Reply:
I tend to keep it simple here because no one is really using its value and I'm not satified with original `LroMetadata` so I try to avoid copying its properties. I added `kind` just to make the struct not empty. `kind` is from weidong's suggestion. This is a copy of the core library `kind` which is a string: https://github.com/Azure/typespec-azure/blob/dfeee05c8fe3ec61ff28daeb0db59942ed08f8fe/packages/typespec-azure-core/src/lro-helpers.ts#L183

Reply:
We can gradually improve `SdkLroServiceMetadata`, per people's demand.

Reply:
It is actually me that suggested to make it a type (that can have more properties later), than an enum.

This is in case that we need more info in finalStep, in future.
But, yes, it probably would involve some break anyway (e.g. if we later make the type polymorphic).

Reply:
I would say if it's a type in the original definition instead of an enum, I lean towards keeping it a type